tests/conftest.py

```diff
@@ -14,15 +14,60 @@ def montage():
 
 @pytest.fixture(scope="session")
 def raw():
-    """Fixture for physionet EEG subject 4, dataset 1."""
+    """Return an `mne.io.Raw` object for use with unit tests.
+
+    This fixture downloads and reads in subject 4, run 1 from the Physionet
+    BCI2000 (eegbci) open dataset. This recording is quite noisy and is thus a
+    good candidate for testing the PREP pipeline.
+
+    File attributes:
+    - Channels: 64 EEG
+    - Sample rate: 160 Hz
+    - Duration: 61 seconds
+
+    This is only run once per session to save time downloading.
+
+    """
     mne.set_log_level("WARNING")
-    # load in subject 1, run 1 dataset
+
+    # Download and read S004R01.edf from the BCI2000 dataset
     edf_fpath = eegbci.load_data(4, 1, update_path=True)[0]
+    raw = mne.io.read_raw_edf(edf_fpath, preload=True)
+    eegbci.standardize(raw)  # Fix non-standard channel names
+
+    return raw
+
 
-    # using sample EEG data (https://physionet.org/content/eegmmidb/1.0.0/)
+@pytest.fixture(scope="session")
+def raw_clean(montage):
+    """Return an `mne.io.Raw` object with no bad channels for use with tests.
+
+    This fixture downloads and reads in subject 1, run 1 from the Physionet
+    BCI2000 (eegbci) open dataset, interpolates its two bad channels (T10 & F8),
+    and performs average referencing on the data. Intended for use with tests
+    where channels are made artifically bad.
+
+    File attributes:
+    - Channels: 64 EEG
+    - Sample rate: 160 Hz
+    - Duration: 61 seconds
+
+    This is only run once per session to save time downloading.
+
+    """
+    mne.set_log_level("WARNING")
+
+    # Download and read S001R01.edf from the BCI2000 dataset
+    edf_fpath = eegbci.load_data(1, 1, update_path=True)[0]
     raw = mne.io.read_raw_edf(edf_fpath, preload=True)
+    eegbci.standardize(raw)  # Fix non-standard channel names
+
+    # Interpolate the file's few bad channels to produce a clean dataset
+    raw.set_montage(montage)
+    raw.info["bads"] = ["T10", "F8"]
+    raw.interpolate_bads()
 
-    # The eegbci data has non-standard channel names. We need to rename them:
-    eegbci.standardize(raw)
+    # Re-reference the data after interpolating bad channels
+    mne.set_eeg_reference(raw, 'average', copy=False, ch_type='eeg')
```

Comment:
I honestly don't know, I haven't really used black formatting before! If you have a preference for double quotes over singles for this project I'll definitely stick to that.

Reply:
From the look of it, PyPREP doesn't have any sort of automatic linting for Black enabled other than the 88 line length, is that something that should be fixed?

And that make sense! My general habit to use single-quotes for data (e.g., variable names, argument strings, key names, etc.) and double-quotes for strings that are meant to be read as text (e.g. docstrings, print messages, exception messages, etc.), but I'm happy to conform to whatever conventions a project uses. From what I can tell the Black style guide [prefers double-quotes to singles](https://black.readthedocs.io/en/stable/the_black_code_style.html#strings) (like you thought), so up to you.

Reply:
argh, good point actually. This line is supposed to ensure in CI that all code is BLACK formatted: https://github.com/sappelhoff/pyprep/blob/bc46978b912737bc3b843438a3cbe1cd9d4dc33d/.github/workflows/python_tests.yml#L41

But it should say `black . --check` instead of `black .` --> the former does a check, whereas the latter just applies the formatting, which is then obviously lost because it's not committed.

And for devs, the development environment should be setup automatically such that BLACK formatting is applied prior to each commit (pre-commit hook, see https://github.com/sappelhoff/pyprep#installation)

I'll make a PR for that.